smooth_pose_traj/src/smooth_pose_traj.cpp

```diff
@@ -34,177 +51,170 @@ geometry_msgs::Pose SmoothPoseTraj::interpPose(const geometry_msgs::Pose& P1,
   P.orientation.y = (1 - alpha) * P1.orientation.y + alpha * P2.orientation.y;
   P.orientation.z = (1 - alpha) * P1.orientation.z + alpha * P2.orientation.z;
   P.orientation.w = (1 - alpha) * P1.orientation.w + alpha * P2.orientation.w;
-  if (qnormalize(P) == false)
-  {
-    printf("normalize failed inside intermpPose()\n");
-  }
+
+  // Normalize Quaternion
+  qnormalize(P);
+
   return (P);
 }
 geometry_msgs::Pose SmoothPoseTraj::getNPtAveragePose(const geometry_msgs::PoseArray& input_poses,
                                                       int pose_index,
                                                       int n_pts)
 {
-  size_t n = input_poses.poses.size();
-  geometry_msgs::Pose P;
-  P.position.x = 0;
-  P.position.y = 0;
-  P.position.z = 0;
-  P.orientation.x = 0;
-  P.orientation.y = 0;
-  P.orientation.z = 0;
-  P.orientation.w = 0;
+  int n = static_cast<int>(input_poses.poses.size());
+  geometry_msgs::Pose p;
+  p.position.x = 0;
+  p.position.y = 0;
+  p.position.z = 0;
+  p.orientation.x = 0;
+  p.orientation.y = 0;
+  p.orientation.z = 0;
+  p.orientation.w = 0;
 
   if (pose_index == 0 || pose_index == n - 1)
-    return (input_poses.poses[pose_index]);
+    return (input_poses.poses[static_cast<std::size_t>(pose_index)]);
 
   int total = 0;
-  size_t n_each_side = n_pts / 2;
+  int n_each_side = n_pts / 2;
   int start = pose_index - n_each_side;
   int stop = pose_index + n_each_side;
   if (start < 0)
     start = 0;
   if (stop > n)
     stop = n;
-  for (size_t i = start; i < stop; i++)
+  for (size_t i = static_cast<size_t>(start); i < static_cast<size_t>(stop); i++)
```

Comment:
here, I would have simply used for(int i = sart; i<stop; i++), why cast? Does the index need to be size_t?

Reply:
This was to get rid of warning because start and stop are of type `int` which was implicitly being casted to `size_t`.